### DISCUSSION TITLE
How does using Feathers to make existing REST API real time work? Does it poll?

### DISCUSSION THREAD

**Discussion by gohanko** [feathersjs/feathers#3551] [Q&A] [answered]
Hi Daffl and anybody reading,

I'm in the research phase of building a public transportation backend API that tracks the real time location of the vehicles (bus, trains).

The data source for this is a real time API made available by my government. The layer I want to implement is a layer that helps dispenses these data in real time without burdening the original service. Essentially, we call the original API, cache the data / do some manipulation, then send it out in real time to active clients. Effectively, it means the original API only has one client connected instead of thousands.

I've come across [this issue](https://github.com/feathersjs/feathers/issues/501) on which the [linked article](https://blog.feathersjs.com/using-feathersjs-to-make-your-existing-api-real-time-813f7c3fd892#.ave1gtmq8) by @daffl helps clarify some stuff on how to do so. What I learned is that I can create a service to connect to an external API and Feathers will handle the real time part.

My question is:

1. How does Feathers ensure the data is real time? Does it do some kind of polling? If the original service updates something, how would Feathers know?
2. Is it possible for me to set the interval for this polling? The data source updates every 30 seconds, so maybe I can make it similar.

Kind Regards,
Brandon Y.

**Comment by daffl** [comment, marked as answer]
Automatic real-time functionality will only work when the Feathers API is also used to modify the data. If the data changes externally you will have to implement your own mechanism to get the changed data and send the appropriate real-time events.

**Reply by gohanko** [reply]
Thanks @daffl. One way I'm thinking is to build the API using feathers, and have an endpoint that runs code that gets the data. Then we'll have a basic script that calls that endpoint per x time. This way, data propagation using Feathers API would work without any external solutions.

**Reply by daffl** [reply]
Correct. If you call the other services in that endpoint, their real-time events will be sent out automatically to all connected clients.
